metpy/calc/tools.py

```diff
@@ -831,3 +832,296 @@ def _less_or_close(a, value, **kwargs):
+    delta_diff = delta[delta_slice1] - delta[delta_slice0]
+    center = (- delta[delta_slice1] / (combined_delta * delta[delta_slice0]) * f[slice0] +
+              delta_diff / (delta[delta_slice0] * delta[delta_slice1]) * f[slice1] +
+              delta[delta_slice0] / (combined_delta * delta[delta_slice1]) * f[slice2])
+
+    # Fill in "left" edge with forward difference
+    slice0[axis] = slice(None, 1)
+    slice1[axis] = slice(1, 2)
+    slice2[axis] = slice(2, 3)
+    delta_slice0[axis] = slice(None, 1)
+    delta_slice1[axis] = slice(1, 2)
+
+    combined_delta = delta[delta_slice0] + delta[delta_slice1]
+    big_delta = combined_delta + delta[delta_slice0]
+    left = (- big_delta / (combined_delta * delta[delta_slice0]) * f[slice0] +
+            combined_delta / (delta[delta_slice0] * delta[delta_slice1]) * f[slice1] -
+            delta[delta_slice0] / (combined_delta * delta[delta_slice1]) * f[slice2])
+
+    # Now the "right" edge with backward difference
+    slice0[axis] = slice(-3, -2)
+    slice1[axis] = slice(-2, -1)
+    slice2[axis] = slice(-1, None)
+    delta_slice0[axis] = slice(-2, -1)
+    delta_slice1[axis] = slice(-1, None)
+
+    combined_delta = delta[delta_slice0] + delta[delta_slice1]
+    big_delta = combined_delta + delta[delta_slice1]
+    right = (delta[delta_slice1] / (combined_delta * delta[delta_slice0]) * f[slice0] -
+             combined_delta / (delta[delta_slice0] * delta[delta_slice1]) * f[slice1] +
+             big_delta / (combined_delta * delta[delta_slice1]) * f[slice2])
+
+    return concatenate((left, center, right), axis=axis)
+
+
+@exporter.export
+def second_derivative(f, **kwargs):
+    """Calculate the second derivative of a grid of values.
+
+    Works for both regularly-spaced data and grids with varying spacing.
+
+    Either `x` or `delta` must be specified.
+
+    Either `x` or `delta` must be specified. This uses 3 points to calculate the
+    derivative, using forward or backward at the edges of the grid as appropriate, and
+    centered elsewhere. The irregular spacing is handled explicitly, using the formulation
+    as specified by [Bowen2005]_.
+
+    Parameters
+    ----------
+    f : array-like
+        Array of values of which to calculate the derivative
+    axis : int, optional
+        The array axis along which to take the derivative. Defaults to 0.
+    x : array-like, optional
+        The coordinate values corresponding to the grid points in `f`.
+    delta : array-like, optional
+        Spacing between the grid points in `f`. There should be one item less than the size
+        of `f` along `axis`.
+
+    Returns
+    -------
+    array-like
+        The second derivative calculated along the selected axis.
+
+    See Also
+    --------
+    first_derivative
+
+    """
+    n, axis, delta = _process_deriv_args(f, kwargs)
+
+    # create slice objects --- initially all are [:, :, ..., :]
+    slice0 = [slice(None)] * n
```

Comment:
Given that these 10-ish lines are duplicated in the first derivative, should they be in a helper somewhere? Trivial lines, but more to maintain/bugfix and makes the functions even longer.

Reply:
I thought about this, but they each have their own set of 5 slices that would have to be passed in--that seemed like it wouldn't simplify enough to justify.